### PR TITLE
feat: allow additional properties to be added as a dictionary of keys

### LINKED ISSUE #1652: Additional properties should be added to the `Form`
Any additional properties not specified by the JSON schema should still be displayed by the `Form` if they have been specified using the `additionalProperties` keyword. See: https://json-schema.org/understanding-json-schema/reference/object.html#properties

This does not include the `additionalProperties` being a boolean.

This requires the UI to be able to add arbitrarily named property keys, similar to the handling of "children". Equally similarly if the items are in data (and not defaults) then they should be removable. See also the "children" UI.

### PR DESCRIPTION
**Pull request by janechu**
# Description

Deciphers the `additionalProperties` keyword in JSON schemas to create a dictionary of properties based on the definition included in the keyword.

This allows:
- Adding a new key
- Modifying a key
- Removing a key

~The "soft remove" functionality inherent in all form controls allows for the removal of dictionary items.~ Soft remove is disabled for dictionary keys.

## Motivation & context

Closes #1652 

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-tooling-react/src/form/form/form-item.dictionary.tsx [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+    }
+
+    private renderFormControls(): React.ReactNode {
+        return (typeof this.props.data !== "undefined"
+            ? Object.keys(this.props.data)
+            : []
+        )
+            .filter((dataKey: string) => {
+                return !this.props.enumeratedProperties.includes(dataKey);
+            })
```
Comment (nicholasrice) [inline packages/fast-tooling-react/src/form/form/form-item.dictionary.tsx]: You could just use reduce and avoid multiple loops on the array.

just saying...

File: packages/fast-tooling-react/src/form/form/form-section.tsx [hunk 2]
```diff
@@ -359,11 +400,23 @@ class FormSection extends React.Component<
...
                 </div>
             </div>
         );
     }
 
+    /**
+     * Get all enumerated properties for the object
+     */
+    private getEnumeratedProperties(schema: any): string[] {
+        if (typeof schema.properties === "undefined") {
```
Comment (nicholasrice) [inline packages/fast-tooling-react/src/form/form/form-section.tsx]: you could probably just check by value here - `schema.properties === undefined`

File: packages/fast-tooling-react/src/style/utilities.ts [hunk 3]
```diff
@@ -292,6 +292,45 @@ export function applyRemoveItemStyle(): CSSRules<{}> {
...
+            background: foreground300,
+        },
+        "&::after": {
+            position: "absolute",
+            content: "''",
+            pointerEvents: "none",
+            width: "1px",
+            height: "9px",
+            left: "9.5px",
+            top: "5.5px",
```
Comment (nicholasrice) [inline packages/fast-tooling-react/src/style/utilities.ts]: `"&::before, &::after"` will save some duplication here

File: packages/fast-tooling-react/src/form/form/form-item.dictionary.tsx (hunk 1 above)
Comment (janechu) [reply packages/fast-tooling-react/src/form/form/form-item.dictionary.tsx]: Darn it

**Pull request merged**